### ISSUE TITLE
New braille context presentation modes sometimes break panning backwards

### ISSUE THREAD

**Issue by dkager** [nvaccess/nvda#7468]
### Steps to reproduce:

1. Open the NVDA Log Viewer (NVDA+n, T, V).
2. Pan braille backwards.

NB: The context presentation mode in use doesn't matter.

### Expected behavior:

The braille dispay shows "NVDA Log Viewer multi line ro edt".

### Actual behavior:

Nothing happens.

### System configuration:

NVDA version: master-14282,1ce6e944
I traced this to commit 1785d5f32 (merged #7236).

### Other questions:

So far reproduced in wx edit controls, Notepad, WordPad, Word, Notepad++.

CC @leonardder

**Comment by zstanecic** [comment]
i can comfirm that using the latest snapshot and handytech braille wave display without bluetooth from source.

**Comment by derekriemer** [comment]
@leonardder

**Comment by leonardder** [comment]
Ugh, let me investigate this either today or tomorrow, but as soon as 
possible anyhow.

**Comment by leonardder** [comment]
There is another issue which already existed before this new context presentation functionality.

### Steps to reproduce
1. Open de log viewer, and position the caret at the first line
2. Pan backwards. The line starts with "NVDA Log Viewer mln ro edt".
3. Pan forward
4. Pan backward ones again

Expected: The line starts with "NVDA Log Viewer mln ro edt"
Actual: You have to pan back another time, as focusToHardLeft has been applied for the multi line edit field, so the start position has been restricted again, similar to how the state of the display was after step 1.

This issue gets a bit more prominent with the new context presentation options implementation, since in some cases, multiple regions have the focusToHardLeft property set to True. This shouldn't have been a problem since this property should only have effect on the last region, but in the case of panning backwards, it clearly has side effects.

I think I'd be able to fix this properly in the following way:

1. Ignore everything regarding focusToHardLeft when panning around, only apply it when a new object gains focus
2. Make sure only one region has focusTohardLeft set to True, even though point 1 makes it slightly unnecessary to fix this.

I don't think that this should be a part of the pr for this particular issue and it shouldn't go straight into master, as the issue itself is very trivial. Again, note that this issue already existed for ages, and it seems it has never been considered major.

@JCSTeh: thoughts?

**Comment by jcsteh** [comment]
> Actual: You have to pan back another time, as focusToHardLeft has been applied for the multi line edit field, so the start position has been restricted again, similar to how the state of the display was after step 1.

While I guess it's asymmetric, I'd argue this is potentially useful/correct. It means you can always get back to the state you were in when the control gained focus, as opposed to focus being entirely special.

> 2. Make sure only one region has focusTohardLeft set to True, even though point 1 makes it slightly unnecessary to fix this.

Would this improve the state of things if you didn't implement point 1?

**Comment by leonardder** [comment]
[**@jcsteh**](https://github.com/jcsteh) commented on [3 aug. 2017 01:38 CEST](https://github.com/nvaccess/nvda/issues/7468#issuecomment-319827805 "2017-08-02T23:38:44Z - Replied by Github Reply Comments"):
> > Actual: You have to pan back another time, as focusToHardLeft has been applied for the multi line edit field, so the start position has been restricted again, similar to how the state of the display was after step 1.
> 
> While I guess it's asymmetric, I'd argue this is potentially useful/correct. It means you can always get back to the state you were in when the control gained focus, as opposed to focus being entirely special.

Mmm. Wonder what @DKager thinks about this, especially now we have a special script to move the display to the current focus.

> > 2. Make sure only one region has focusTohardLeft set to True, even though point 1 makes it slightly unnecessary to fix this.
> 
> Would this improve the state of things if you didn't implement point 1?

Sure. For example, in the python console

1. Open the console and input random data
2. Press f6 to move to the output area, and focus the first line
3. Pan backwards

Expected: The display reads: "NVDA Python Console mln"
Actual: The display starts with mln

Actually, I think I might be able to fix this in one line of code. Furthermore, it will make it a lot easier to provide proper unit tests for #7470.
